readthedocs/search/parse_json.py

```diff
@@ -1,77 +1,89 @@
 """Functions related to converting content into dict/JSON structures."""
 
-import json
 import logging
+import orjson as json
 
 from django.conf import settings
 from django.core.files.storage import get_storage_class
 
-from pyquery import PyQuery
+from selectolax.parser import HTMLParser
 
 
 log = logging.getLogger(__name__)
 
 
-def generate_sections_from_pyquery(body, fjson_storage_path):
-    """Given a pyquery object, generate section dicts for each section."""
+def generate_page_sections(body, fjson_storage_path):
+    """Generate section dicts for each section."""
 
     # Removing all <dl> tags to prevent duplicate indexing with Sphinx Domains.
+    nodes_to_be_removed = []
     try:
         # remove all <dl> tags which contains <dt> tags having 'id' attribute
-        dt_tags = body('dt[id]')
-        dt_tags.parents('dl').remove()
+        dt_tags = body.css('dt[id]')
+        for tag in dt_tags:
+            parent = tag.parent
+            if parent.tag == 'dl':
+                nodes_to_be_removed.append(parent)
     except Exception:
         log.exception('Error removing <dl> tags from file: %s', fjson_storage_path)
 
-    # remove toctree elements
+    # remove `Table of Contents` elements
     try:
-        body('.toctree-wrapper').remove()
+        nodes_to_be_removed += body.css('.toctree-wrapper') + body.css('.contents.local.topic')
```

Comment:
It seems there is a logic change here. Are we removing an extra node here?

Reply:
Yes, we are removing extra node here.
Earlier, I thought that all the `Table of Contents` are under a div having class `.toctree-wrapper` but latest I realised that pages which have `Table of Contents` describing the contents in the same page (like: [SEO Guide](https://docs.readthedocs.io/en/stable/guides/technical-docs-seo-guide.html#technical-documentation-search-engine-optimization-seo-guide)) are having different css class -- `contents local topic`.
So I included that also.